pkg/identities/service.go

```diff
@@ -0,0 +1,224 @@
+	defer span.End()
+
+	identities, rr, err := s.kratos.ListIdentitiesExecute(
+		s.buildListRequest(ctx, page, size, credID),
+	)
+
+	data := new(IdentityData)
+
+	if err != nil {
+		s.logger.Error(err)
+		data.Error = s.parseError(rr)
+	}
+
+	data.Identities = identities
+
+	// TODO @shipperizer check if identities is defaulting to empty slice inside kratos-client
+	if data.Identities == nil {
+		data.Identities = make([]kClient.Identity, 0)
+	}
+
+	return data, err
+}
+
+func (s *Service) GetIdentity(ctx context.Context, ID string) (*IdentityData, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.IdentityApi.GetIdentity")
+	defer span.End()
+
+	identity, rr, err := s.kratos.GetIdentityExecute(
+		s.kratos.GetIdentity(ctx, ID),
+	)
+
+	data := new(IdentityData)
+
+	if err != nil {
+		s.logger.Error(err)
+		data.Error = s.parseError(rr)
+	}
+
+	if identity != nil {
+		data.Identities = []kClient.Identity{*identity}
+	} else {
+		data.Identities = []kClient.Identity{}
+	}
+
+	return data, err
+}
+
+func (s *Service) CreateIdentity(ctx context.Context, bodyID *kClient.CreateIdentityBody) (*IdentityData, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.IdentityApi.CreateIdentity")
+	defer span.End()
+
+	if bodyID == nil {
+		err := fmt.Errorf("no identity data passed")
+
+		data := new(IdentityData)
+		data.Identities = []kClient.Identity{}
+		data.Error = s.parseError(nil)
+		data.Error.SetMessage(err.Error())
+
+		s.logger.Error(err)
+
+		return data, err
+	}
+
+	identity, rr, err := s.kratos.CreateIdentityExecute(
+		s.kratos.CreateIdentity(ctx).CreateIdentityBody(*bodyID),
+	)
+
+	data := new(IdentityData)
+
+	if err != nil {
+		s.logger.Error(err)
+		data.Error = s.parseError(rr)
+	}
+
+	if identity != nil {
+		data.Identities = []kClient.Identity{*identity}
+	} else {
+		data.Identities = []kClient.Identity{}
+	}
+
+	return data, err
+}
+
+func (s *Service) UpdateIdentity(ctx context.Context, ID string, bodyID *kClient.UpdateIdentityBody) (*IdentityData, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.IdentityApi.UpdateIdentity")
+	defer span.End()
+	if ID == "" {
+		err := fmt.Errorf("no identity ID passed")
+
+		data := new(IdentityData)
+		data.Identities = []kClient.Identity{}
+		data.Error = s.parseError(nil)
+		data.Error.SetMessage(err.Error())
+
+		s.logger.Error(err)
+
+		return data, err
+	}
+
+	if bodyID == nil {
+		err := fmt.Errorf("no identity body passed")
+
+		data := new(IdentityData)
+		data.Identities = []kClient.Identity{}
+		data.Error = s.parseError(nil)
+		data.Error.SetMessage(err.Error())
+
+		s.logger.Error(err)
+
+		return data, err
+	}
+
+	identity, rr, err := s.kratos.UpdateIdentityExecute(
+		s.kratos.UpdateIdentity(ctx, ID).UpdateIdentityBody(*bodyID),
+	)
+
+	data := new(IdentityData)
+
+	if err != nil {
+		s.logger.Error(err)
+		data.Error = s.parseError(rr)
+	}
+
+	if identity != nil {
+		data.Identities = []kClient.Identity{*identity}
```

Comment:
Why return a list here when it's always one item?

Reply:
I was never a fun of the AnyOf in the Jason schemas so returning a list is a decent approach as it gives a reliable response

arguable point I know, happy to switch it to a single object but worth checking with UI team